### PR TITLE
Parsing fixes and code reorganization

### PR DESCRIPTION
**Pull request by williamvwang**
- Fix parsing broken by ESPN changes
- Reorganize code
    - Move startup procedure and team mapping to `__init__`
    - Move difflib fuzzy lookup to `__init__` as a lambda
    - Add `utils.py`

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import requests
+import difflib
+from .utils import unescape
```
Comment (rjchee) [inline __init__.py]: TIL you can import like that

File: utils.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import re, html.entities
+
+##
+# http://effbot.org/zone/re-sub.htm#unescape-html
+#
+# Removes HTML or XML character references and entities from a text string.
+#
+# @param text The HTML (or XML) source text.
+# @return The plain text, as a Unicode string, if necessary.
+def unescape(text):
```
Comment (rjchee) [inline utils.py]: There's no standard library function for unescaping HTML?

File: __init__.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+def getTeams():
+    teamMap = {}
+    try:
+        resp = requests.get('http://www.espn.com/college-football/teams').text
```
Comment (rjchee) [inline __init__.py]: Maybe we should add a timeout to the request? Apparently requests will hang forever if the server doesn't respond.

File: app.py [hunk 4]
```diff
@@ -40,8 +20,8 @@ def getDate(body):
 
 @app.route('/<teamname>')
 def getTeamStats(teamname):
-    teamname = difflib.get_close_matches(teamname.lower(), teamIds.keys(), 1, 0.2)[0]
-    team_id, teamname = teamIds[teamname]
+    teamname = fuzzymatch(teamname.lower(), 1)[0]
+    team_id, teamname = TeamsByName[teamname]
     for year in range(datetime.now().year, 2001, -1):
         resp = requests.get('http://www.espn.com/college-football/team/schedule/_/id/' + team_id + '/year/' + str(year)).text
```
Comment (rjchee) [inline app.py]: timeout here too

**Review by rjchee — CHANGES_REQUESTED** [review]
requests.get(url, timeout=n)

File: __init__.py (hunk 1 above)
Comment (williamvwang) [reply __init__.py]: same

File: utils.py (hunk 2 above)
Comment (williamvwang) [reply utils.py]: [u right](https://docs.python.org/3/library/html.html#html.unescape)

**Review by rjchee — APPROVED** [review]

**Pull request merged**